src/main/proto/com/google/api/codegen/config.proto

```diff
@@ -270,13 +272,19 @@ message MethodConfigProto {
   // IAM methods to IamPolicy for Pub/Sub.
   string reroute_to_grpc_interface = 12;
 
+  // Specifies the list of the method's input message fields (including nested
+  // ones, using the dot notation), which should be url-encoded in
+  // "x-goog-request-params" header
```

Comment:
I think we should also have the header key configurable.

Reply:
This header is hardcoded on google server side. It is analogous to url path and url parameters in HTTP/1+rest request. In other words it is not just a header, it is a support for url-path/params in https/2+grpc world. The name of the header is not going to change, if we will ever want to add more headers, they will be something different (i.e. they will be legit headers, instead of being url-path imitation for grpc).